include/mbedtls/rsa.h

```diff
@@ -74,7 +75,173 @@ extern "C" {
 #endif
 
 /**
- * \brief          RSA context structure
+ * Helper functions for RSA-related operations on MPI's.
+ */
+
+/**
+ * \brief          Compute RSA prime moduli P, Q from public modulus N=PQ
+ *                 and a pair of private and public key.
+ *
+ * \note           This is a 'static' helper function not operating on
+ *                 an RSA context. Alternative implementations need not
```

Comment:
@yanesca So notes for alternative implementations should be in plain comments only?

Reply:
@hanno-arm As far as I can tell the notes and references in question fall into two categories:
- notes that can be deduced from the position of the corresponding preprocessor guards and I think we can safely omit these notes.
- instructions to/statements about alternative implementations in the description. Alternative implementations have to implement our interface and have to comply to ours in everything anyway, because it is prescribed in `config.h`. Therefore I think we can safely rephrase these for the sake of clarity, to talk about the function in general and not to mention alternative implementations.

If there is anything that does not fall into the above two category, it should go to either `config.h` into a comment above the flag, like with the ECP module replacement, or to its own section in the corresponding Mbed Handbook page.

Reply:
@gilles-peskine-arm I agree that it can be useful, but my main concerns are:
- that this additional information can be confusing for the reader, makes the documentation much longer and thus increases the risk of being simply ignored
- If we mix them now, then it is going to be a lot of work to separate them when we'll actually decide to have separate documentation for platform builders.

Reply:
I also think information that is primarily intended for alt-implementers has its place in the doxygen documentation. I agree with Janos that applications developers should be the primary target of the doxygen documentation, but where I differ is I don't think this prevents us from having also information for other audiences, as long as it's introduced in a way that doesn't cause unnecessary confusion to the primary audience. (And as I commented above, I think some parts of the newly-introduced documentation as it stands are not clear enough in that respect, but that can be improved.)

Anyway, maybe this is a broader discussion that we should have outside this PR in order to establish general guidelines?

Reply:
Manuel's solution seems a reasonable compromise between the position taken by Gilles and Hanno and mine, and I am happy to take this compromise in order to get this PR merged as soon as possible.

I also would be happy to have a broader discussion outside this PR to establish general guidelines.